api/swagger.yaml

```diff
@@ -0,0 +1,788 @@
+  version: 1.0.0
+externalDocs:
+  description: Find out more about Space Tycoon game
+  url: https://github.com/gdg-garage/space-tycoon
+servers:
+- url: https://space-tycoon.garage-trip.cz/
+tags:
+- name: login
+  description: Get user session
+- name: staticdata
+  description: Static game state (planets)
+- name: data
+  description: "Dynamic game data (scores, prices, spaceship positions)"
+- name: commands
+  description: Specify commands for your ships
+- name: tick
+  description: Signal that your turn is over for the current tick
+paths:
+  /login:
+    post:
+      tags:
+      - login
+      requestBody:
+        content:
+          '*/*':
+            schema:
+              $ref: '#/components/schemas/Credentials'
+        required: true
+      responses:
+        "200":
+          description: OK
+          content: {}
+        "403":
+          description: Invalid name or password
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        "405":
+          description: Invalid input
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /staticdata:
+    post:
+      tags:
+      - staticdata
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/StaticData'
+  /data:
+    post:
+      tags:
+      - data
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Data'
+  /commands:
+    post:
+      tags:
+      - commands
+      requestBody:
+        content:
+          '*/*':
+            schema:
+              $ref: '#/components/schemas/Commands'
+        required: true
+      responses:
+        "200":
+          description: OK
+          content: {}
+        "405":
+          description: Invalid input
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /tick:
+    post:
+      tags:
+      - tick
+      requestBody:
+        content:
+          '*/*':
+            schema:
+              $ref: '#/components/schemas/Tick'
+        required: true
+      responses:
+        "200":
+          description: OK
+          content: {}
+        "405":
+          description: Invalid input
+          content: {}
+components:
+  schemas:
+    Command:
+      required:
+      - type
+      type: object
+      properties:
+        type:
+          type: string
+      discriminator:
+        propertyName: Type_
+      oneOf:
+      - $ref: '#/components/schemas/MoveCommand'
+      - $ref: '#/components/schemas/StopCommand'
+      - $ref: '#/components/schemas/ConstructCommand'
+      - $ref: '#/components/schemas/DecommissionCommand'
```

Comment:
Hmm, I don't see a reason why not. It can give people ability to change strategy mid game. I don't see any possible abuse because even with getting half your money back it's expensive to do this.

We might see someone rush to build a "shipyard" and sell their mothership to get a lot of money at once. This assumes that there is a cheap enough "shipyard". And I don't really think we mind people doing this.

Reply:
moving to separate issue: https://github.com/gdg-garage/space-tycoon/issues/26